python/paddle/fluid/tests/unittests/mkldnn/test_matmul_mkldnn_op.py

```diff
@@ -161,5 +161,231 @@ def set_attributes(self):
         self.attrs = {'force_fp32_output': True}
 
 
+@skip_check_grad_ci(reason="DNNL's MatMul doesn't implement grad kernel.")
+class TestMatMulOpReshapeTranspose(OpTest):
+    def generate_data(self):
+        self.x = np.random.random([2, 128, 768]).astype("float32").reshape(
+            [2, 128, 12, 64]).transpose([0, 2, 1, 3])
+        self.y = np.random.random([2, 128, 768]).astype("float32").reshape(
+            [2, 128, 12, 64]).transpose([0, 2, 1, 3])
+        self.out = np.matmul(self.x, self.y.transpose([0, 1, 3, 2]))
+        self.shape_x = []
+        self.axis_x = []
+        self.shape_y = []
+        self.axis_y = []
+
+    def setUp(self):
+        # Set max isa, otherwise fails on SKX and earlier
+        os.environ["DNNL_MAX_CPU_ISA"] = "AVX"
```

Comment:
Do we need to check if the machine support avx and then set it, if not support give some information to user.

Reply:
I copied this from Michal's test to be safe https://github.com/PaddlePaddle/Paddle/pull/23754/files/eda126700bd34bd0650ff57999ccc4fd04a4d5de#diff-68b25bf1ad071a1bf85630c689dd071bR36